CMakeLists.txt

```diff
@@ -264,114 +440,130 @@ install(
     COMPONENT headers
 )
 
-include(ExternalProject)
-
-# include(mergestaticlibs)
 # ------------------------------------------------------------
 # Enable OpenMP support?
 # ------------------------------------------------------------
-if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    option(OPENMP_ENABLE "Enable openMP support" OFF)
-else()
-    option(OPENMP_ENABLE "Enable openMP support" ON)
-endif()
-
-if(OPENMP_ENABLE)
-    # message(STATUS "****** finding OpenMP support")
-    find_package(OpenMP)
+if(ENABLE_OPENMP)
+    find_package(OpenMP QUIET)
     if(OPENMP_FOUND)
-        if(NOT ${CMAKE_VERSION} VERSION_LESS "3.9")
-            link_libraries(OpenMP::OpenMP_C OpenMP::OpenMP_CXX)
-        else()
-            add_compile_options(
-                $<$<COMPILE_LANGUAGE:C>:${OpenMP_C_FLAGS}>
-                $<$<COMPILE_LANGUAGE:CXX>:${OpenMP_CXX_FLAGS}>
-            )
-            link_libraries(${OpenMP_C_FLAGS})
-        endif()
+        # TODO this needs to apply to each thing that might want openmp
+        # TODO also the COMPILE_LANGUAGE generator was in cmake as of 3.3, why was it conditional on 3.9?
```

Comment:
Because findOpenMP targets were enabled in cmake 3.9 before then only flags and lists were available .  It is preferable to use the defined target if available

Reply:
I got this backwards, yeah